sql/dd/impl/upgrade/server.cc

```diff
@@ -986,21 +984,24 @@ static const char *percona_telemetry_install[] = {
     "PREPARE stmt FROM @str;\n",
     "EXECUTE stmt;\n",
     "DROP PREPARE stmt;\n",
-    "UPDATE mysql.user SET Repl_slave_priv = 'Y' WHERE User = 'mysql.session' "
-    "AND Host = 'localhost';\n",
-    "UPDATE mysql.user SET Repl_client_priv = 'Y' WHERE User = 'mysql.session' "
-    "AND Host = 'localhost';\n",
-    "INSERT IGNORE INTO mysql.tables_priv VALUES ('localhost', 'mysql', "
-    "'mysql.session', 'component', 'root@localhost', NOW(), 'Select', '');\n",
-    "INSERT IGNORE INTO mysql.tables_priv VALUES ('localhost', "
-    "'performance_schema', 'mysql.session', 'replication_group_members', "
-    "'root@localhost', NOW(), 'Select', '');\n",
+    "INSERT IGNORE INTO mysql.user VALUES "
+    "('localhost','percona.telemetry','N','N','N','N','N','N','"
+    "N','N','N','N','N','N','N','N','N','N','N','N','N','N','N','N','N','N','N'"
+    ",'N','N','N','N','','','','',0,0,0,0,'caching_sha2_password','$A$005$"
+    "THISISACOMBINATIONOFINVALIDSALTANDPASSWORDTHATMUSTNEVERBRBEUSED','N',"
+    "CURRENT_TIMESTAMP,NULL,'Y', 'N', 'N', NULL, NULL, NULL, NULL);\n",
+    "UPDATE mysql.user SET Select_priv = 'Y', Repl_slave_priv = 'Y', "
+    "Repl_client_priv = 'Y' WHERE User = 'percona.telemetry' AND Host = 'localhost';\n",
```

Comment:
Could you please elaborate why do we need UPDATE as well? Perhaps extend the comment to explain this?

Reply:
Ok. I guess this is to handle the case when user manually revokes the privileges from percona.telemtry. IMO still worth commenting.

Reply:
I've added more clarifications in the code comment.